docs/devbook/upgrade-crates-and-openapi-versions/upgrade_crates_and_openapi_versions.sh

```diff
@@ -0,0 +1,101 @@
+readonly ORANGE="\e[1;33m"
+readonly GREEN="\e[1;32m"
+readonly RESET="\e[0m"
+
+readonly OPEN_API_FILE=openapi.yaml
+declare OPEN_API_UPDATE=""
+declare OPEN_API_UPDATE_MESSAGE=""
+
+update_crate_versions() {
+    local -r dry_run=$1
+    local -r -n files_modify=$2
+    local -r -a members="$(cargo get workspace.members --delimiter " ")"
+    local -i nb_files_modify=0
+    local package_name
+
+    local cargo_options=""
+    if [ true = "$dry_run" ]
+    then
+        cargo_options=--dry-run
+    fi
+
+    for member in $members
+    do
+        nb_files_modify=$(echo "$files_modify" | grep -c "^$member/")
+        if [[ $nb_files_modify -gt 0 ]]
+        then
+            package_name=${member##*/}
+            cargo set-version $cargo_options --package "${package_name##*/}" --bump patch
+        fi
+
+    done
+}
+
+update_openapi_version() {
+    local -r dry_run=$1
+    local -r version_line=$(grep -E "version: [0-9]+\.[0-9]+\.[0-9]+" $OPEN_API_FILE)
+    local -r patch_number=$(echo "$version_line" | cut -d . -f 3)
+    local -r next_patch_number=$((patch_number + 1))
+    local -r new_version=$(echo "$version_line" | cut -d . -f 1-2).$next_patch_number
+
+    echo -e "   ${GREEN}Upgrading${RESET} $OPEN_API_FILE from ${version_line##*version: } to ${new_version##*version: }"
+    if [ true = "$dry_run" ]
+    then
+        echo -e "${ORANGE}warning${RESET}: aborting $OPEN_API_FILE update due to dry run"
+    else
+        sed -i "s/$version_line/version: $new_version/g" $OPEN_API_FILE
+    fi
+    OPEN_API_UPDATE="\n* $OPEN_API_FILE from \`${version_line##*version: }\` to \`${new_version##*version: }\`"
+    OPEN_API_UPDATE_MESSAGE=" and \`$OPEN_API_FILE\` version"
+}
+
+################
+readonly COMMAND="${1:-""}"
+readonly COMMIT_REF="HEAD"
+FILES_MODIFY="$(git diff "$COMMIT_REF" --name-only origin/main)"
+readonly -a FILES_MODIFY
+
+declare DRY_RUN=true
+if [[ "$COMMAND" ==  "--run" ]]
+then
+    DRY_RUN=false
+fi
+
+update_crate_versions $DRY_RUN FILES_MODIFY
+
+if [ "$(echo "${FILES_MODIFY[@]}" | grep -xc "$OPEN_API_FILE")" -gt 0 ]
+then
+    update_openapi_version $DRY_RUN
+fi
+
+if [ true = $DRY_RUN ]
+then
+    echo -e "${ORANGE}warning${RESET}: script is run in dry mode. To execute the changes, run ${GREEN}./update_crate_versions.sh --run${RESET}"
+else
+  UPDATED_CRATES="\n$(find . -name Cargo.toml -exec git diff "$COMMIT_REF" {} + | grep -E "^[\+\-]version = \"[0-9\.]+\"|name = " | tr '\n' ' ' | sed -r "s/ name = \"([a-z\-]+)\" -version = \"([0-9\.]+)\" \+version = \"([0-9\.]+)\" ?/* \1 from \`\2\` to \`\3\`\n/g")"
+  COMMIT_MESSAGE=$(echo -e "chore: upgrade crate versions${OPEN_API_UPDATE_MESSAGE}\n${UPDATED_CRATES}${OPEN_API_UPDATE}")
+
+  echo -e "$COMMIT_MESSAGE"
+
+  # Do we want to commit the changes ?
+  #git add $OPEN_API_FILE Cargo.lock ./*/Cargo.toml ./internal/*/Cargo.toml ./mithril-test-lab/*/Cargo.toml && git commit -m "$COMMIT_MESSAGE"
```

Comment:
Maybe this command can also part of the output?

Reply:
I added a `--commit` argument to execute the commit instead.